src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.cs

```diff
@@ -59,6 +64,12 @@ private static async Task InitializeStandardComponentServicesAsync(HttpContext h
             authenticationStateProvider.SetAuthenticationState(Task.FromResult(authenticationState));
         }
 
+        var formData = httpContext.RequestServices.GetRequiredService<FormDataProvider>() as IHostEnvironmentFormDataProvider;
+        if (handler != null && formData != null)
+        {
+            formData.SetFormData(handler, form!.ToDictionary(kvp => kvp.Key, kvp => kvp.Value[0]).AsReadOnly());
```

Comment:
Is there a reason we need to allocate another two dictionaries here rather than just flow `IFormCollection` directly?

Reply:
Yes, `FormDataProvider` lives in `Microsoft.AspNetCore.Components`, so we can't reference it directly.

I'm doing more work in this area in the near future. Right now, I am just getting the things to flow so that we can demo it on preview4.

Reply:
I've iterated a bit on this, after talking to Steve, I've moved this to Microsoft.AspNetCore.Components.Web, and I've taken a dependency there on Microsoft.Extensions.Primitives which is the bit that brings in StringValues.

I still need to wrap IFormCollection sadly as it does not implement IReadOnlyDictionary, I'll follow up to see if that's intentional, an oversight, or something we just never had the request for yet.

An alternative will be to make a separate interface in Components that closely matches IFormCollection. This is something that needs to live in `Microsoft.AspNetCore.Components.Web` as we want RCLs to be able to target it.

As we make progress on the unified project model and its impact (multi-targeting), we might revisit this, so take this with a grain of salt.